kratos/input_output/stl_io.cpp

```diff
@@ -36,6 +37,33 @@ StlIO::StlIO(std::filesystem::path const& Filename)
     mpInputStream = pFile;
 }
 
+StlIO::StlIO(std::filesystem::path const& Filename, const Flags Options)
+    : mOptions(Options)
+{
+    std::fstream* pFile = new std::fstream();
+    std::fstream::openmode OpenMode = std::fstream::in;
+
+    // Set the mode
+    if (mOptions.Is(IO::READ)) {
+        OpenMode = std::fstream::in;
+    } else if (mOptions.Is(IO::APPEND)) {
+        OpenMode = std::fstream::in | std::fstream::app;
+    } else if (mOptions.Is(IO::WRITE)) {
```

Comment:
I don't think writing functionality belongs in this class. All other member functions are read-oriented and none are protected from running in the other mode. E.g opening a file with the WRITE flag and calling read functions.

So it triggers a design question. Should this be added to the class and made safe or added to a separate class for writing only?

Reply:
This class derives from IO, that allows writing and reading, so I think it is perfectly normal that this class can write too (its implicit in the name also). The other member methods are private so they can be called witihin the class, so I think this is safe enough as you are only leaving this decission to the class itself. What do you think as the original designers @pooyan-dadvand  @loumalouomega ?

Reply:
I would do as ModelPartIO